test/with_dummyserver/test_https.py

```diff
@@ -249,6 +390,27 @@ def test_assert_invalid_fingerprint(self):
         https_pool.assert_fingerprint = 'AA'
         self.assertRaises(SSLError, https_pool.request, 'GET', '/')
 
+    def test_assert_invalid_fingerprint_context(self):
+        ctx = create_urllib3_context(cert_reqs=ssl.CERT_REQUIRED)
+        ctx.load_verify_locations(DEFAULT_CA)
+        https_pool = HTTPSConnectionPool('127.0.0.1', self.port,
+                                         ssl_context=ctx)
+
+        https_pool.assert_fingerprint = 'AA:AA:AA:AA:AA:AAAA:AA:AAAA:AA:' \
+                                        'AA:AA:AA:AA:AA:AA:AA:AA:AA'
+
+        self.assertRaises(SSLError, https_pool.request, 'GET', '/')
+        https_pool._get_conn()
+
+        # Uneven length
+        https_pool.assert_fingerprint = 'AA:A'
+        self.assertRaises(SSLError, https_pool.request, 'GET', '/')
+        https_pool._get_conn()
```

Comment:
I copied that from the test above, must have been debugging cruft. I took it out of both methods in ea3633f, and the tests still pass.